From 971ec8a5c6229dc45b89105196a86333635fc553 Mon Sep 17 00:00:00 2001
From: Paul Elder <paul.elder@ideasonboard.com>
Date: Mon, 24 Aug 2020 17:14:29 +0900
Subject: [PATCH] staging: vc04_services: ISP: Fix dmabuf error check
 in S_CTRL

In bcm2835_isp_s_ctrl, the error check for dma_buf_get() is incorrect,
and considers ERR_PTR pointers as valid dmabufs. Fix this error check.

Signed-off-by: Paul Elder <paul.elder@ideasonboard.com>
---
 drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c
+++ b/drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c
@@ -733,7 +733,7 @@ static int bcm2835_isp_s_ctrl(struct v4l
 		       sizeof(struct bcm2835_isp_lens_shading));
 
 		dmabuf = dma_buf_get(ls.dmabuf);
-		if (!dmabuf)
+		if (IS_ERR_OR_NULL(dmabuf))
 			return -EINVAL;
 
 		ret = vc_sm_cma_import_dmabuf(dmabuf,
